src/main/java/bio/terra/service/DrsService.java

```diff
@@ -61,21 +64,26 @@ public DrsService(StudyDao studyDao,
     public DRSObject lookupObjectByDrsId(String drsObjectId) {
         DrsId drsId = parseAndValidateDrsId(drsObjectId);
 
-        FSObject fsObject = fileService.lookupFSObject(
+        FSObjectBase fsObject = fileService.lookupFSObject(
             drsId.getStudyId(),
-            drsId.getFsObjectId(),
-            FSObject.FSObjectType.FILE);
-        return drsObjectFromFSObject(fsObject, drsId.getDatasetId());
+            drsId.getFsObjectId());
+        if (fsObject.getObjectType() != FSObjectType.FILE) {
```

Comment:
if this only works for files, should the method name be lookupFileByDrsId?

but i also thought the DRS spec was going to move to one lookup for either files or dirs. is this an anticipation of that? if so, maybe add that as a comment somewhere (maybe in yaml?)

Reply:
This will all get rewritten. The code is a way for it to limp along until we update it to DRS V1.0. I added a TODO at the top of the module to document the state of the code.